Pybranch/pacman.py

```diff
@@ -25,172 +26,482 @@ def __init__(self):
         self.rect.x = random.randint(self.frame_min, self.frame_max)
         self.rect.y = random.randint(self.frame_min, self.frame_max)
 
-    def dot_creation(self, dot_sprite: pygame.sprite.Group, every_sprite: pygame.sprite.Group):
+
+class DotCreator:
+    def __init__(self) -> None:
```

Comment:
Az egyes osztályoknál, ha egy másik osztályt is felhasználsz benne, szerintem érdemesebb ezeket az osztály paraméterévé tenni, ahogy a mintakódban is szerepelnek. Ebben az esetben például:
```suggestion
    def __init__(self,
                 collide: Collide,
                 bool: ListToBool) -> None:
        self.collide = collide
        self.bool = bool
```
Ehhez a jelenlegi kódstruktúra szerint szükség van arra, hogy mindig az egyes osztályok előtt definiáld azokat az osztályokat, amelyre az hivatkozik. Tehát ebben az esetben a Collide és a ListToBool osztályokat a DotCreator előtt kell létrehozni.